protocol/flows/handshake/handshake.go

```diff
@@ -109,7 +112,7 @@ func HandleHandshake(context HandleHandshakeContext, router *routerpkg.Router,
 
 	err = router.RemoveRoute([]wire.MessageCommand{wire.CmdVersion, wire.CmdVerAck})
 	if err != nil {
-		panic(err)
+		return nil, false, err
```

Comment:
it should panic here

Reply:
We spoke about this and decided to leave it as is.